### DISCUSSION TITLE
[WARNING DPL-0004/5/6] Warning Details

### DISCUSSION THREAD

**Discussion by dineshannayya** [The-OpenROAD-Project/OpenROAD#754] [Q&A]
Can any one give more details on the these WARNING  & also how to solve these issue.

[WARNING DPL-0004] Placed in rows check failed (4).
[WARNING DPL-0005] Overlap check failed (1).
[WARNING DPL-0006] Site check failed (4).

I am trying to do detailed placement using openlane flow and 99% of block are hard core, only 32 cells are there at top-level
There is good amount of free space available. still tool fails with these error.

**Comment by maliberty** [comment]
It is a check at the end of placement to make sure the result is non-overlapping.  It suggests a placement failure but its hard to say anything more without a test case.  Can you provide one?

**Comment by dineshannayya** [comment]
The project had 99% hard macro with less than 32 other cells and there was
more than 30% free space. Still openroad detailed route was placing
overlapping cells with hard macros & giving DPL-4/5/6 warning.

Finally I noticed the issue was resolved after changing
diamond_search_hight more than *250*.

Can give more details on how increased  diamond_search_hight value helped
here ?

**Comment by maliberty** [comment]
I would guess that global placement left cells very far from any legal location.  You might look at a post global-placement def to see.
